lib/mix/tasks/swagger.generate.ex

```diff
@@ -36,32 +40,34 @@ defmodule Mix.Tasks.Phx.Swagger.Generate do
       args,
       switches: [router: :string, help: :boolean],
       aliases: [r: :router, h: :help])
-
+    
     cond do
       (Keyword.get(switches, :help)) ->
         usage()
-      has_no_endpoint() ->
+      has_no_endpoint(switches) ->
         IO.puts "Skipping app #{app_name()}, no Endpoint configured."
       true ->
-        router = load_router(switches)
+        router = router_module(switches)
+        endpoint = endpoint_module(switches)
```

Comment:
What do you think about moving this line out of the `cond` then pass the `endpoint` directly into `has_no_endpoint`?